Add more comprehensive tests on DagCards

stuart23 opened the pull request:
<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->

`data-testid`'s are added and used for tests to ensure the DagCards render as expected. `aria-label`'s are also added to visual indicators to indicate state.

Times are rendered in local timezone so the components are wrapped in the tests to force GMT.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [airflow-core/newsfragments](https://github.com/apache/airflow/tree/main/airflow-core/newsfragments).

stuart23 committed:
Add labels and test-ids

stuart23 committed:
Linter

pierrejeambrun reviewed (commented):
Thanks for the PR.

Thats a good idea to add such tests. 

For now those tests are not successful can you fix them? Also you'll need to run static checks again to make the CI happy. (You can find more information  here for static checks https://github.com/apache/airflow/blob/main/contributing-docs/08_static_code_checks.rst, and here for react tests https://github.com/apache/airflow/blob/main/contributing-docs/15_node_environment_setup.rst

bbovenzi reviewed (commented):
Good idea on adding missing aria-labels.

I would prefer we look up tests via displayed text when possible and only use data-testid when needed. For instance, schedule should just be a text lookup.

stuart23 committed:
Fix TypeScript non-null assertion warning in DagCard test

- Replace non-null assertion with proper null check and array destructuring
- Add explicit error handling for missing mock data
- Add Jest DOM types import for TypeScript compatibility
- Maintains test functionality while following linting rules

stuart23 committed:
Force GMT timezone on timestamps
